### PR TITLE
Add proper hashcode/equals methods to our matchers

### PR DESCRIPTION
**Pull request by tylerbenson**
I copied over the methods that would have been generated had the `HashCodeAndEqualsPlugin` actually been working.

Also updated the matcher tests to use the TypePool for more realistic values.

### CODE REVIEW

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/LoggingFailSafeMatcher.java [hunk 1]
```diff
@@ -52,6 +50,26 @@ public boolean matches(final T target) {
 
   @Override
   public String toString() {
-    return "safeMatcher(try(" + matcher + ") or " + fallback + ")";
+    return "failSafe(try(" + matcher + ") or " + fallback + ")";
```
Comment (mar-kolya) [inline dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/LoggingFailSafeMatcher.java]: Should this be `loggingFailSafeMatcher` or something similar?

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/LoggingFailSafeMatcher.java [hunk 2]
```diff
@@ -14,7 +13,6 @@
  * @see net.bytebuddy.matcher.FailSafeMatcher
  */
 @Slf4j
-@HashCodeAndEqualsPlugin.Enhance
```
Comment (mar-kolya) [inline dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/LoggingFailSafeMatcher.java]: Is there really no way to make this work through gradle somehow?

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/LoggingFailSafeMatcher.java [hunk 3]
```diff
@@ -52,6 +50,26 @@ public boolean matches(final T target) {
 
   @Override
   public String toString() {
-    return "safeMatcher(try(" + matcher + ") or " + fallback + ")";
+    return "failSafe(try(" + matcher + ") or " + fallback + ")";
+  }
+
+  @Override
+  public boolean equals(final Object var1) {
```
Comment (mar-kolya) [inline dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/LoggingFailSafeMatcher.java]: `var1`?

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/LoggingFailSafeMatcher.java (hunk 2 above)
Comment (tylerbenson) [reply dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/LoggingFailSafeMatcher.java]: probably, but not immediately obvious.  Plus AFAIK it wouldn't work in an IDE.

**Pull request merged**